decisions.md

```diff
@@ -0,0 +1,91 @@
+- Reproducibility: Supports Docker, Singularity, and 3 other types of containers.
+- Portable: Can run locally, Slurm, SGE, PBS, and cloud (Google, Kubernetes, and AWS).
+- Continuous checkpoints: Each process in the workflow is checkpointed. It is possible to retry 
+failed workflows and start from the last checkpoint.
+- Supports various scripting languages including Bash, Python, Perl, and others.
+- Enables separation between configuration (how to do) and workflow logic (what to do).
+- Modularization of tasks possible via workflow, sub-workflows, and processes.
+- Provides detailed logs and various types of execution reports.
+
+### How is the NF script structured?
+
+The NF script contains the following structures:
+#### 2.1 DSL and Parameters
+#### 2.2 Definition of processes
+#### 2.3 Definition of workflows
+#### 2.4 Main workflow
+
+Check this source code example: [seq_ocrd_wf_many.nf](https://github.com/MehmedGIT/OPERANDI_TestRepo/blob/master/ExampleWorkflows/Nextflow/workflow4/seq_ocrd_wf_many.nf)
+
+TODO: I will provide more structure-related details here based on the example above.
+
+### Which features of NF do we use, i.e. what features have to be implemented in potential implementations?
+The minimally used features for local runs are the parameters, processes, process decorators, and workflows. I will provide further answers to any following questions related to this main question. I am not sure what else to cover here for now.
+
+## How does parallelization work, both within works and across works?
+
+A Nextflow workflow script contains several processes. Processes are executed independently and are isolated from each other (i.e. they do not have a shared memory space). Communication between the processes is possible only through data channels (similar to the pipes model in Unix). These channels are basically asynchronous FIFO queues. Any process can define one or more channels as input and output. The order of interaction between these processes, and ultimately the order of workflow execution depends on the communication channel dependencies between processes. For example, if process A writes data to channel A and process B reads data from channel A, then Nextflow knows that process A must be executed before process B.
+
+Check this source code example: [seq_ocrd_wf_many.nf](https://github.com/MehmedGIT/OPERANDI_TestRepo/blob/master/ExampleWorkflows/Nextflow/workflow4/seq_ocrd_wf_many.nf)
+
+TODO: I will provide more parallelization details here based on the example above.
+
+## How does the NF script interact with the processing server?
+There is still no running processing server. More details will be announced once there is more to talk about. The interaction will most probably happen with curl through a bash script inside the Nextflow process. Of course, if it is integrated inside the OCR-D core, then no direct interactions will be needed from inside the Nextflow script.

```

Comment:
Yes, most probably curl if we decide to use just shell scripts inside the execution block of a process. We also have other options by using the Groovy language but this would make the scripts unnecessarily complex. 

"No direct interactions [...] needed" - Having more thought about it, yes you are right. Then NF will be just used to make requests to the processing server and gather the provided responses. Is that right? If so, we kind of lose the advantage of NF to provide proper resource usage reports. The part of the answer after "Of course, [...]" will be removed.